frontend/public/components/instantiate-template.tsx

```diff
@@ -198,6 +201,14 @@ class TemplateForm_ extends React.Component<
       metadata: {
         generateName: `${obj.data.metadata.name}-`,
         namespace,
+        ownerReferences: [
+          {
+            apiVersion: secret.apiVersion,
+            kind: secret.kind,
+            name: secret.metadata.name,
+            uid: secret.metadata.uid,
+          },
+        ],
```

Comment:
I think ownerReferences needs to be set on the secret if I understand correctly https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents

Reply:
Thanks @rottencandy.   

It worked! The secret is getting deleted along with the Template Instance